### ISSUE TITLE
useQuery returns undefined on first render when data is in cache

### ISSUE THREAD

**Issue by BramKaashoek** [apollographql/apollo-client#6686]
I use a piece of code high up in my component tree as such: 
```
  const { data: } = useQuery(GET_CART);
  if (!data.cart) return null;
  return <>{props.children}</>;
```
This used to mean that all children ALWAYS had data available on useQuery(GET_CART), since it would be cached.
This makes it a lot easier for me to do destructuring or use this data in other hooks in child components.

Since updating from apollo client 2 to @apollo/client 3 (3.0.2 to be exact)  a useQuery(GET_CART) will ALWAYS return an `undefined` first  in the child components before returning the data from the cache.

Is there a way to not get the first `undefined` on initial render?

**Comment by benjamn** [comment]
Yep, this is a known issue, and will be fixed in `@apollo/client@3.1.0` (forthcoming). Feel free to head over to #6659 to continue the discussion. Closing now, but happy to reopen if I'm wrong about the sameness of these issues.

**Comment by benjamn** [comment]
@BramKaashoek I believe this issue is fixed in `@apollo/client@3.1.0-pre.3`, thanks to #6710, if you want to give it a try.

**Comment by benjamn** [comment]
Following up: we just published `@apollo/client@3.1.0` to npm!

**Comment by dan-cooke** [comment]
@benjamn 
This is occurring again on `3.5.6` going to raise a bug
